contracts/proxies/StakingProxy.sol

```diff
@@ -0,0 +1,30 @@
+pragma solidity ^0.4.24;
+
+import "../Staking.sol";
+import "./ThinProxy.sol";
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+
+
+contract StakingProxy is ThinProxy {
+    // keccak256("aragon.network.staking")
+    bytes32 internal constant IMPLEMENTATION_SLOT = 0xbd536e2e005accda865e2f0d1827f83ec8824f3ea04ecd6131b7c10058635814;
+
+    string internal constant ERROR_WRONG_IMPLEMENTATION_SLOT = "STAKING_PROXY_WRONG_IMPL_SLOT";
+
+    constructor(Staking _implementation, ERC20 _token) ThinProxy(address(_implementation)) public {
+        bytes4 selector = _implementation.initialize.selector;
+        bytes memory initializeData = abi.encodeWithSelector(selector, _token);
+        bool success = address(_implementation).delegatecall(initializeData);
+
+        assembly {
+            let output := mload(0x40)
+            mstore(0x40, add(output, returndatasize))
+            returndatacopy(output, 0, returndatasize)
```

Comment:
All of this only needs to be executed if the initialize call reverted, we can save some gas by only doing the the copy if we are going to use it for the revert